packages/main/src/plugin/docker/docker-context-handler.spec.ts

```diff
@@ -0,0 +1,202 @@
+
+describe('getCurrentContext', () => {
+  test('should return default if docker config does not exist', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(false);
+
+    const currentContext = await dockerContextHandler.getCurrentContext();
+    expect(currentContext).toBe('default');
+  });
+
+  test('should return context if docker config exist', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+    vi.spyOn(fs.promises, 'readFile').mockResolvedValue(JSON.stringify({ currentContext: 'test-context' }));
+
+    const currentContext = await dockerContextHandler.getCurrentContext();
+    expect(currentContext).toBe('test-context');
+  });
+
+  test('should return undefined if docker config exist but fails to parse it', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+    vi.spyOn(fs.promises, 'readFile').mockResolvedValue('not a JSON');
+
+    const currentContext = await dockerContextHandler.getCurrentContext();
+    expect(currentContext).toBe('default');
+    expect(console.error).toBeCalledWith('Error parsing docker config file', expect.any(Error));
+  });
+});
+
+describe('switchContext', () => {
+  test('should report an error if file does not exists', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(false);
+
+    await expect(() => dockerContextHandler.switchContext('foo')).rejects.toThrowError('does not exist');
+  });
+
+  test('should throw error if context does not exists', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+
+    vi.spyOn(dockerContextHandler, 'getContexts').mockResolvedValue([]);
+
+    await expect(() => dockerContextHandler.switchContext('foo')).rejects.toThrowError('Context foo not found');
+  });
+
+  test('should write the content', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+    vi.spyOn(dockerContextHandler, 'getContexts').mockResolvedValue([
+      { name: 'foo' } as unknown as DockerContextParsingInfo,
+      { name: 'bar' } as unknown as DockerContextParsingInfo,
+    ]);
+    vi.spyOn(fs.promises, 'readFile').mockResolvedValue(JSON.stringify({ currentContext: 'bar' }));
+    await dockerContextHandler.switchContext('foo');
+    // check using the correct indentation
+    expect(fs.promises.writeFile).toBeCalledWith(
+      expect.any(String),
+      JSON.stringify({ currentContext: 'foo' }, null, '\t'),
+    );
+  });
+
+  test('should throw error if JSON is invalid', async () => {
+    vi.spyOn(fs, 'existsSync').mockReturnValue(true);
+    vi.spyOn(dockerContextHandler, 'getContexts').mockResolvedValue([
+      { name: 'foo' } as unknown as DockerContextParsingInfo,
+      { name: 'bar' } as unknown as DockerContextParsingInfo,
+    ]);
+    vi.spyOn(fs.promises, 'readFile').mockResolvedValue('not a json');
+
+    await expect(() => dockerContextHandler.switchContext('foo')).rejects.toThrowError(
+      'Error parsing docker config file',
+    );
+  });
+});
+
+describe('listContexts', () => {
```

Comment:
Missing test for list or getContexts if folder does not exist.

Reply:
docker-context-handler.ts           |     100 |      100 |     100 |     100 |

100% coverage in this file now